src/CSET/operators/collapse.py

```diff
@@ -163,6 +163,13 @@ def collapse_by_hour_of_day(
             # have effectively done this.
             cube.remove_coord("forecast_reference_time")
 
+        collapsed_cubes.append(cube)
+
+    # Retain only common time points between the different models.
+    collapsed_cubes = get_common_time_cubes(collapsed_cubes)
```

Comment:
```suggestion
    collapsed_cubes = collapsed_cubes.extract_overlapping("time")
```
Can we just use [CubeList.extract_overlapping](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris.cube.html#iris.cube.CubeList.extract_overlapping) here, and avoid needing the `get_common_time_cubes` function at all?